sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureDirectoryStream.java

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.models.BlobListDetails;
+import com.azure.storage.blob.models.ListBlobsOptions;
+
+import java.io.IOException;
+import java.nio.file.DirectoryIteratorException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * An object for iterating the contents of a directory.
+ *
+ * This type does not validate that the given path is a directory; a listing call will be made against the given
+ * resource in either case, and if the resource is a file, it will simply return 0 results (assuming there are no
```

Comment:
Files.newDirectoryStream throws `NotDirectoryException` `if the file could not otherwise be opened because it is not a directory` . Should we be consistent with that?

Reply:
It's listed as an optional specific exception, so I was going to save checking the status, but can make a call to ensure it's a directory.